store/task.go

```diff
@@ -745,7 +756,15 @@ func (s *Store) patchTaskStatusImpl(ctx context.Context, tx *sql.Tx, patch *api.
 			return nil, FormatError(err)
 		}
 		taskPatchedRaw = &taskRaw
+		found = true
+	}
+	if err := row.Err(); err != nil {
+		return nil, FormatError(err)
+	}
+	if !found {
```

Comment:
On line 658, we check the task existence. How do we get to not found case again?

Reply:
Also consider to use different row names so that they can do defer Close() independently.

Reply:
> On line 658, we check the task existence. How do we get to not found case again?

I think you are right. In a repeated read transaction, this will never happen. I'll remove the `found` check here.

Reply:
Changed to using `QueryRowContext()` to use the `database/sql`'s built-in handling of single-row query. It's much cleaner now.